src/Propel/Runtime/ActiveQuery/Criteria.php

```diff
@@ -186,7 +186,7 @@ class Criteria
     protected $joins = [];
 
     /**
-     * @var \Propel\Runtime\ActiveQuery\Criteria[]
+     * @var array<string,\Propel\Runtime\ActiveQuery\Criteria>
```

Comment:
Do all tools and IDEs understand this newer syntax?
Or should those also go into phpstan-var here?

Reply:
Thank you! I'm already so familiar with documenting `array<keytype,valuetype>` that I completely forgot that it is non-standard phpdoc. No-one can ever attest that all tools understand some nonstandard syntax, although I know from my own and colleagues' experience that some of the big ones (JetBrains PHPstorm, many plugins for Visual Studio Code, Sublime) all support it well.
I'll fix this and re-submit the PR!

Reply:
... and done!

To keep this PR manageable, I'll leave the template annotations for generated classes for a later time. With the changes as presented at least the results of many queries will provide more insight for static analysis tools.